plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSink.java

```diff
@@ -462,7 +459,8 @@ private FileWriter createParquetFileWriter(String path)
         try {
             Closeable rollbackAction = () -> fileSystem.deleteFile(path);
 
-            List<Type> parquetTypes = dataColumnTypes.stream()
+            List<Type> parquetTypes = typedDataColumns.stream()
+                    .map(column -> column.type())
                     .map(type -> {
                         if (type instanceof TimestampWithTimeZoneType) {
```

Comment:
this can be also extracted to a utility method

Reply:
outdated